lib/mli_parser.ml

```diff
@@ -0,0 +1,179 @@
+open! Compat
+
+module Code_block = struct
+  type t = { location : Odoc_model.Location_.span; contents : string }
+end
+
+(* drop_first_and_last [1; 2; 3; 4] = [2; 3]. *)
+let drop_first_and_last list = List.tl list |> List.rev |> List.tl |> List.rev
+
+let slice file_contents ~(start : Odoc_model.Location_.point)
+    ~(end_ : Odoc_model.Location_.point) =
+  let lines_to_include =
+    String.split_on_char '\n' file_contents
+    |> List.mapi (fun i line -> (succ i, line))
+    |> List.filter (fun (lnum, _) -> start.line <= lnum && lnum <= end_.line)
+    |> List.map snd
```

Comment:
This has a running time of `number of lines * number of comments`, that's `O(N^2)`.

What do you think of splitting the file into an array in `parse_mli` and accessing lines by their line number in the array ?
Another benefit of this is the computation of `end` in `parse_mli` will be far easier and more robust.

Reply:
@Julow good call, I've done this.